tests/test_component.py

```diff
@@ -0,0 +1,52 @@
+import sys
+from pathlib import Path
+from unittest.mock import MagicMock
+
+import dask.dataframe as dd
+import pytest
+from fondant.component import FondantTransformComponent
+from fondant.dataset import FondantDataset
+
+
+components_path = Path(__file__).parent / "example_specs/components"
+
+
+def test_transform_kwargs(monkeypatch):
+    """Test that arguments are passed correctly to `Component.transform` method."""
+
+    class EarlyStopException(Exception):
+        """Used to stop execution early instead of mocking all later functionality"""
+
+    # Mock `Dataset.load_dataframe` so no actual data is loaded
+    def mocked_load_dataframe(self, spec):
+        return dd.from_dict({"a": [1, 2, 3]}, npartitions=1)
+
+    monkeypatch.setattr(FondantDataset, "load_dataframe", mocked_load_dataframe)
```

Comment:
Just a question, would it be cleaner to use the `mock.patch` decorator function?

https://docs.python.org/3/library/unittest.mock.html

Reply:
There doesn't seem to be a clear choice: https://github.com/pytest-dev/pytest/issues/4576

I like the fact that with `monkeypatch`, the patched code is included inside of your test function, which is not the case with the `mock.patch` decorator.